ape_frame/accounts.py

```diff
@@ -34,37 +52,64 @@ def alias(self) -> str:
     def address(self) -> AddressType:
         return self.web3.eth.accounts[0]
 
-    def sign_message(self, msg: SignableMessage) -> Optional[MessageSignature]:
-        try:
-            raw_signature = self.web3.eth.sign(self.address, hexstr=_hash_eip191_message(msg).hex())
-        except ValueError as e:
-            if not e.args[0]["message"] == "User declined transaction":
-                raise
+    def sign_message(self, msg: Any, **signer_options) -> Optional[MessageSignature]:
+        raw_signature = None
 
-            return None
+        if isinstance(msg, str):
+            raw_signature = wrap_sign(lambda: self.web3.eth.sign(self.address, text=msg))
+        elif isinstance(msg, int):
+            raw_signature = wrap_sign(
+                lambda: self.web3.eth.sign(self.address, hexstr=HexBytes(msg).hex())
+            )
+        elif isinstance(msg, bytes):
+            raw_signature = wrap_sign(lambda: self.web3.eth.sign(self.address, hexstr=msg.hex()))
+        elif isinstance(msg, SignableMessage):
+            raw_signature = wrap_sign(lambda: self.web3.eth.sign(self.address, data=msg.body))
+        elif isinstance(msg, EIP712Message):
+            raw_signature = wrap_sign(
+                lambda: self.web3.eth.sign_typed_data(
+                    self.address, msg._body_  # type: ignore[arg-type]
```

Comment:
@z80dev may know more about the type discrepancy here, but I created this issue: https://github.com/ethereum/web3.py/issues/3255